homeassistant/components/linknlink/config_flow.py

```diff
@@ -0,0 +1,265 @@
+            if device.is_locked:
+                return await self.async_step_unlock()
+            return await self.async_step_finish()
+
+        await self.async_set_unique_id(device.mac.hex())
+        _LOGGER.error(
+            "Failed to authenticate to the device at %s: %s", device.host[0], err_msg
+        )
+        return self.async_show_form(step_id="auth", errors=errors)
+
+    async def async_step_reset(self, user_input=None, errors=None) -> FlowResult:
+        """Guide the user to unlock the device manually.
+
+        We are unable to authenticate because the device is locked.
+        The user needs to open the LinknLink app and unlock the device.
+        """
+        device = self.device
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reset",
+                errors=errors,
+                description_placeholders={
+                    "name": device.name,
+                    "model": device.model,
+                    "host": device.host[0],
+                },
+            )
+
+        return await self.async_step_user(
+            {CONF_HOST: device.host[0], CONF_TIMEOUT: device.timeout}
+        )
+
+    async def async_step_unlock(self, user_input=None) -> FlowResult:
+        """Unlock the device.
+
+        The authentication succeeded, but the device is locked.
+        We can offer an unlock to prevent authorization errors.
+        """
+        device = self.device
+        errors = {}
+
+        if user_input is None:
+            pass
+
+        elif user_input["unlock"]:
+            try:
+                await self.hass.async_add_executor_job(device.set_lock, False)
+
+            except NetworkTimeoutError as err:
+                errors["base"] = "cannot_connect"
+                err_msg = str(err)
+
+            except LinknLinkException as err:
+                errors["base"] = "unknown"
+                err_msg = str(err)
+
+            except OSError as err:
+                if err.errno == errno.ENETUNREACH:
+                    errors["base"] = "cannot_connect"
+                    err_msg = str(err)
+                else:
+                    errors["base"] = "unknown"
+                    err_msg = str(err)
+
+            else:
+                return await self.async_step_finish()
+
+            _LOGGER.error(
+                "Failed to unlock the device at %s: %s", device.host[0], err_msg
+            )
+
+        else:
+            return await self.async_step_finish()
+
+        data_schema = {vol.Required("unlock", default=False): bool}
+        return self.async_show_form(
+            step_id="unlock",
+            errors=errors,
+            data_schema=vol.Schema(data_schema),
+            description_placeholders={
+                "name": device.name,
+                "model": device.model,
+                "host": device.host[0],
+            },
+        )
+
+    async def async_step_finish(self, user_input=None) -> FlowResult:
```

Comment:
If I understand correctly this function now only creates a new entry, so I think it's best to remove the "step" part of this. (don't name it like a step and dont accept user_input)